front/lib/api/assistant/configuration.ts

```diff
@@ -68,194 +92,223 @@ export async function getAgentConfiguration(
   return res[0] || null;
 }
 
-export async function getAgentConfigurations<V extends "light" | "full">({
-  auth,
-  agentsGetView,
-  agentPrefix,
-  variant,
-}: {
-  auth: Authenticator;
-  agentsGetView: AgentsGetViewType;
-  agentPrefix?: string;
-  variant: V;
-}): Promise<
-  V extends "light" ? LightAgentConfigurationType[] : AgentConfigurationType[]
-> {
-  // {agentId: string} view: get a specific agent configuration.
-  // list view: specific to a user. Get all agents that are in the user's list (either they created it and it is private, or via AgentUserRelation).
-  // {conversationId: string} view: specific to a conversation. Get all agents that are in the user's list + workspace/published agents in the current conversation.
-  // all view: workspace + published agents (not private ones), eg. for agent gallery
-  // workspace view: only workspace agents (not published ones)
-  // published view: only published agents (not workspace ones)
-  // global view: only global agents (not workspace or published ones)
-  // admin_internal view: all agents, including private ones => this is only for internal use (eg poke). Requires superuser or admin auth.
-
-  const owner = auth.workspace();
-  if (!owner || !auth.isUser()) {
-    throw new Error("Unexpected `auth` without `workspace`.");
-  }
-  const plan = auth.plan();
-  if (!plan) {
-    throw new Error("Unexpected `auth` without `plan`.");
-  }
+function makeApplySortAndLimit(sort?: SortStrategy, limit?: number) {
```

Comment:
For the now, I've opted to keep everything together because what we have is a simple setup, and it's clear we don't want to build on it as-is. It might be a good idea for us to chat more about sorting options. That way, we can design a sorting algorithm that maps easily with SQL, making pagination free to implement.